src/interpreter/environment.ts

```diff
@@ -0,0 +1,54 @@
+import { BrokenEnvironmentError, UnboundNameError } from './errors';
+import { type Environment, type EnvironmentFrame } from './types/interpreter';
+
+export const extendEnvironment = (
+  names: string[],
+  values: number[],
+  environment: Environment
+): Environment => {
+  if (names.length !== values.length) {
```

Comment:
Actually, is there a reason why this function doesn't just take in some representation where this can never happen?